test/test_core.py

```diff
@@ -7722,6 +7722,41 @@ def test_source_map(self, args):
     else:
       self.assertTrue(seen_lines.issuperset([6, 7, 11, 12]), seen_lines)
 
+  @no_asan("asan does not work with dynamic linking")
+  @needs_dylink
+  def test_embind_dylink_visibility_hidden(self):
+    # Check that embind is usable from a library built with "-fvisibility=hidden"
+    create_file('liba.cpp', r'''
+      #include <emscripten/val.h>
+      #define EXPORT __attribute__((visibility("default")))
+      using namespace emscripten;
+      EXPORT void liba_fun() {
+        unsigned char buffer[1];
+        val view(typed_memory_view(1, buffer));
+      }
+    ''')
+
+    # side module
+    self.build_dlfcn_lib('liba.cpp', outfile='liba.so', emcc_args=['-std=c++11', '-fvisibility=hidden'])
+
+    # main module
+    self.set_setting('MAIN_MODULE', 1)
+    self.clear_setting('SIDE_MODULE')
+
+    self.emcc_args = ['liba.so', '-L.', '--bind']
```

Comment:
Can you move this to `emcc_args=` in the call to do_runf?

Reply:
Yep, done,